### PR TITLE
Update transfer learning tutorial

### PR DESCRIPTION
**Pull request by 8bitmp3**
@MarkDaoust

### CODE REVIEW

File: site/en/tutorials/images/transfer_learning.ipynb [hunk 1]
```diff
@@ -69,7 +69,7 @@
         "id": "hRTa3Ee15WsJ"
```
Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <p>Why remove these? It's the one tutorial where this would be apropriate.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <blockquote>two transfer learning strategies—<em>feature extraction</em> and <em>fine-tuning</em></blockquote><p> </p><p>I'm not sure these are separate enough to talk about as different *strategies*. You use an existing model to extract the features. You build a classifier on top of that. You need to train the classifier. You can fine-tune the freature-extractor.</p><p> </p><blockquote>previously trained on a&nbsp;typically&nbsp;large dataset for a particular task, such as&nbsp;image&nbsp;classification</blockquote><p> </p><p>This doesn't parse correctly.</p><p> </p><blockquote>A pre-trained model is a <a href="https://github.com/tensorflow/docs/raw/19d2699318803eab2c0c88f92cf072da887194fc/site/en/guide/saved_model.ipynb" target="_blank">saved model</a></blockquote><p> </p><p>A pre-trained model can be a saved model, but it could be other things too.</p><p> </p><blockquote>You will classify images of cats and dogs by using pre-trained models from <a href="https://tfhub.dev/" target="_blank">TensorFlow Hub</a></blockquote><p> </p><p>Would we gain anything to parameterize over the datasets (flowers or Cats and Dogs) or the model source (hub or keras-applications)?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <p>Try not to leave empty sections between headings.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <blockquote>Learn more about image preprocessing.</blockquote><p> </p><p>Just like this, it comes across as a command, instead of as an option. "you can learn more there" vs. "learn more!".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <p><code>ds =image_dataset_from_directory(..., shuffle=true)</code>  followed by <code>take/skip</code> doesn't seem right.</p><p> </p><p>I think you need the <code>validation_split</code> argument if you're doing that.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <blockquote>pre-trained models from TensorFlow expect float inputs in the <code>[0, 1]</code> range for image inputs</blockquote><p> </p><p>"from TensorFlow Hub".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <blockquote>normalization_layer</blockquote><p> </p><p>Consider renaming that to rescaling.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <blockquote>"Learn more about <a href="https://www.tensorflow.org/tutorials/images/data_augmentation" target="_blank">data augmentation</a>"</blockquote><p> </p><p>Sounds like a command.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <p>Try to avoid empty sections between headings.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <p>It's a bit funny that the rescaling layer is applied to the dataset, but the data-augmentation isn't.</p><p>We create the <code>data_augmention</code> model, but do't use it here.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='54'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <p> </p><p>It's always better to show the number than write it in the text, so nobody makes a bug if it comes out to be 97 or 99%</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='86'/>

**Comment by MarkDaoust** [comment]
Here's a first round of comments.

**Comment by 8bitmp3** [comment]
Thanks @MarkDaoust . @anirudh161 PTAL! Thanks 👍

File: site/en/tutorials/images/transfer_learning.ipynb (hunk 1 above)
Comment (8bitmp3) [reply site/en/tutorials/images/transfer_learning.ipynb:69]: <p>Done 🐈🐕</p>

Comment (8bitmp3) [reply site/en/tutorials/images/transfer_learning.ipynb:69]: <p>OK, done!</p>

Comment (8bitmp3) [reply site/en/tutorials/images/transfer_learning.ipynb:69]: <p>Rephrased ("For more information, check out the...")</p>

Comment (8bitmp3) [reply site/en/tutorials/images/transfer_learning.ipynb:69]: <p>Done!</p>

Comment (8bitmp3) [reply site/en/tutorials/images/transfer_learning.ipynb:69]: <p>Thanks. Done.</p>

Comment (8bitmp3) [reply site/en/tutorials/images/transfer_learning.ipynb:69]: <blockquote><br></blockquote><blockquote>"two transfer learning strategies—<em>feature extraction</em>&nbsp;and&nbsp;<em>fine-tuning"</em></blockquote><blockquote>I'm not sure these are separate enough to talk about as different *strategies*.</blockquote><p>"two transfer learning approaches". LMKWYT</p>

Comment (8bitmp3) [reply site/en/tutorials/images/transfer_learning.ipynb:75]: There are 2 directories: 1 for training and 1 for validation. Then we take 20% of the validation set and use it for testing.

The `validation_split=0.2, subset=...` arg in `tf.keras.utils.image_dataset_from_directory` only reserves data for training or validation sets, which we already have.

```
`subset` must be either "training" or "validation"
```

Comment (8bitmp3) [reply site/en/tutorials/images/transfer_learning.ipynb:75]: <p>Data augmentation layers have been applied, since:<code>data_augmentation = tf.keras.Sequential([tf.keras.layers.RandomFlip('horizontal'), tf.keras.layers.RandomRotation(0.2),])</code>  </p><p> </p><p>Let's just rewrite the <code>model</code> :</p><p> </p><pre class="ql-syntax" spellcheck="false">model = tf.keras.Sequential([
  data_augmentation,
  new_base_model,
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(num_classes)
]) 
</pre><p> </p><p>Earlier, I added two new sentences spelling out the layer names and descriptions + why we need this <code>data_augmentation</code> model. I also added a link to the KPL guide.</p><p> </p><pre class="ql-syntax" spellcheck="false">### Data augmentation

...

Create a data augmentation model using the <code>tf.keras.layers.RandomFlip</code> and <code>tf.keras.layers.RandomRotation</code> preprocessing layers. The former randomly flips images during training, while the latter—rotates them.

data_augmentation = ...
...

Later, you need to add <code>data_augmentation</code> as a layer inside your main Keras model before the training begins.

Note: These [preprocessing layers](https://www.tensorflow.org/guide/keras/preprocessing_layers) are active only during training when you call Keras <code>Model.fit</code>. They are inactive when the model is used in inference mode in <code>Model.evaluate</code> or <code>Model.fit</code>.

</pre>

Comment (8bitmp3) [reply site/en/tutorials/images/transfer_learning.ipynb:75]: <p>The <code>Model.fit</code> output after 20 epochs and TensorBoard's UI should be enough then. So:</p><p> </p><p>"After fine-tuning, the model should reach a high accuracy on the validation set."</p>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <pre class="ql-syntax" spellcheck="false">Line #1.    result_batch = base_model.predict(train_dataset)
</pre><p> </p><p>Here you want to predict on just one batch, not the whole dataset.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <pre class="ql-syntax" spellcheck="false">Line #2.    image_batch, label_batch = test_dataset.as_numpy_iterator().next()
</pre><p> </p><p>This comes out all black check the scaling of the input image.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='91'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <p>Instead of callking this "base_model" and "new_base_model" could we call it "pretrained_classifier" and "feature_extractor"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <p>Remove the shuffle=True from the validation dataset</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (MarkDaoust) [inline site/en/tutorials/images/transfer_learning.ipynb:69]: <p>Why not do this at the end?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/2081/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='70'/>
